### PR TITLE
mhv account changes to support mhv-signin

### PR DESCRIPTION
**Pull request by saneshark**

### CODE REVIEW

File: app/models/mhv_account.rb [hunk 1]
```diff
@@ -72,6 +80,10 @@ def preexisting_account?
     user&.mhv_correlation_id.present?
```
Comment (patrickvinograd) [inline app/models/mhv_account.rb]: This seems like it needs changing to reflect addition of mhv_uuid.

File: app/models/mhv_account.rb [hunk 2]
```diff
@@ -56,12 +56,20 @@ class MhvAccount < ActiveRecord::Base
...
   end
 
+  # if you already have an MHV account you're eligible to use it on vets.gov
+  # but if you don't have one, we will only create / upgrade if you're a va patient.
   def eligible?
-    va_patient?
+    va_patient? || preexisting_account?
+  end
+
+  def identity_proof_valid?
```
Comment (patrickvinograd) [inline app/models/mhv_account.rb]: This logic should definitely not live in `mhv_account`

File: app/models/mhv_account.rb (hunk 1 above)
Comment (saneshark) [reply app/models/mhv_account.rb]: Yea, so mhv_correlation_id now is basically:

`mhv_uuid || mvi.va_profile.mhv_correlation_id` on user.rb

where mhv_uuid has precedence, there is another method I created `mhv_signin?` which tests/checks for this.

File: app/models/mhv_account.rb (hunk 2 above)
Comment (saneshark) [reply app/models/mhv_account.rb]: I agree, I mostly put it in here to call attention to it. I don't think it hurts to have this check in here for logic readability, but yes, it probably ought to be a method on user.rb perhaps to replace any validations we remove.

**Comment by saneshark** [comment]
This is stale.

**Pull request closed without merging**